### DISCUSSION TITLE
Coarsening Vector Potential

### DISCUSSION THREAD

**Discussion by aborderies** [idefix-code/idefix#241] [Q&A] [answered]
Hi all ! 

I am currently trying to restart some of the simulations I've carried out, but with better resolution. To do this, I'm trying to interpolate the field values on the new grid. However, in some parts of my grid, I use the coarsening module, and to interpolate the value of the different fields correctly, I need to know what the module does to each field. For density and velocity it's fine, but for vector potential (in my case only the AX3 component as my simulation is in 2.5D) I haven't found in the source code how it is treated when the grid is coarsened. 
Is it possible to explain what the grid coarsening module does to the AX3 field/how this field is treated by idefix when the grid coarsening is activate ? 

Thanks in advance for the help.

**Comment by glesur** [comment, marked as answer]
Hi @Anto6453 , so when using the vector potential formulation, you have to see the vector potential as a temporal integral of the EMF over time. This is always true, with or without grid coarsening.

When grid coarsening is used, B is computed from A on the normal grid, and only then, coarsening is applied to B. In practice, this means that only A on the coarsened grid corners is effectively used.

In other words, it probably means that you can interpolate naively A on your new grid, and it will work out of the box.

**Reply by aborderies** [reply]
Ok I see, thanks for the explanation!
